x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/GetApiKeyRequestTests.java

```diff
@@ -100,4 +113,40 @@ public void writeTo(StreamOutput out) throws IOException {
             }
         }
     }
+
+    public void testSerialization() throws IOException {
+        final String apiKeyId = randomAlphaOfLength(5);
+        final boolean ownedByAuthenticatedUser = true;
+        GetApiKeyRequest getApiKeyRequest = GetApiKeyRequest.usingApiKeyId(apiKeyId, ownedByAuthenticatedUser);
+        {
+            ByteArrayOutputStream outBuffer = new ByteArrayOutputStream();
+            OutputStreamStreamOutput out = new OutputStreamStreamOutput(outBuffer);
+            out.setVersion(Version.V_7_2_0);
+            getApiKeyRequest.writeTo(out);
+
+            InputStreamStreamInput inputStreamStreamInput = new InputStreamStreamInput(new ByteArrayInputStream(outBuffer.toByteArray()));
+            inputStreamStreamInput.setVersion(randomFrom(Version.V_7_2_0, Version.V_7_3_0));
+            GetApiKeyRequest requestFromInputStream = new GetApiKeyRequest(inputStreamStreamInput);
+
+            assertThat(requestFromInputStream.getApiKeyId(), equalTo(getApiKeyRequest.getApiKeyId()));
+            // old version so the default for `ownedByAuthenticatedUser` is false
+            assertThat(requestFromInputStream.ownedByAuthenticatedUser(), is(false));
+        }
+        {
+            ByteArrayOutputStream outBuffer = new ByteArrayOutputStream();
+            OutputStreamStreamOutput out = new OutputStreamStreamOutput(outBuffer);
+            out.setVersion(randomFrom(Version.V_7_4_0, Version.CURRENT));
+            getApiKeyRequest.writeTo(out);
+
+            InputStreamStreamInput inputStreamStreamInput = new InputStreamStreamInput(new ByteArrayInputStream(outBuffer.toByteArray()));
+            inputStreamStreamInput.setVersion(randomFrom(Version.V_7_4_0, Version.CURRENT));
+            GetApiKeyRequest requestFromInputStream = new GetApiKeyRequest(inputStreamStreamInput);
+
+            assertThat(requestFromInputStream, equalTo(getApiKeyRequest));
+        }
+    }
+
+    private static String randomNullOrEmptyString() {
+        return randomFrom(new String[]{"", null});
```

Comment:
It's personal preference, but ...
```suggestion
        return randomBoolean() ? "" : null;
```